lldb/source/ValueObject/DILEval.cpp

```diff
@@ -613,29 +638,38 @@ Interpreter::Visit(const ArraySubscriptNode *node) {
     return child_valobj_sp;
   }
 
-  int64_t signed_child_idx = node->GetIndex();
+  int64_t signed_child_idx = idx->GetValueAsSigned(0);
   return base->GetSyntheticArrayMember(signed_child_idx, true);
 }
 
 llvm::Expected<lldb::ValueObjectSP>
 Interpreter::Visit(const BitFieldExtractionNode *node) {
-  auto lhs_or_err = Evaluate(node->GetBase());
-  if (!lhs_or_err)
-    return lhs_or_err;
-  lldb::ValueObjectSP base = *lhs_or_err;
-  int64_t first_index = node->GetFirstIndex();
-  int64_t last_index = node->GetLastIndex();
+  auto base_or_err = EvaluateAndDereference(node->GetBase());
+  if (!base_or_err)
+    return base_or_err;
+  lldb::ValueObjectSP base = *base_or_err;
+  auto first_idx_or_err = EvaluateAndDereference(node->GetFirstIndex());
+  if (!first_idx_or_err)
+    return first_idx_or_err;
+  lldb::ValueObjectSP first_idx = *first_idx_or_err;
+  auto last_idx_or_err = EvaluateAndDereference(node->GetLastIndex());
+  if (!last_idx_or_err)
+    return last_idx_or_err;
+  lldb::ValueObjectSP last_idx = *last_idx_or_err;
+
+  if (!first_idx->GetCompilerType().IsIntegerOrUnscopedEnumerationType() ||
+      !last_idx->GetCompilerType().IsIntegerOrUnscopedEnumerationType()) {
+    return llvm::make_error<DILDiagnosticError>(
+        m_expr, "bit index is not an integer", node->GetLocation());
+  }
+
+  int64_t first_index = first_idx->GetValueAsSigned(0);
+  int64_t last_index = last_idx->GetValueAsSigned(0);
```

Comment:
We should probably check the success value (i.e., pass it the `bool` output parameter). Otherwise we'd return some range that the user didn't ask for